subsystem/cluster_test.go

```diff
@@ -42,6 +42,31 @@ const (
 	minSuccessesInRow = 2
 )
 
+// The order of this list matters. It needs to follow a cluster happy-flow
+var clusterValidSortedStatuses = [...]string{
+	models.ClusterStatusInsufficient,
+	models.ClusterStatusReady,
+	models.ClusterStatusPreparingForInstallation,
+	models.ClusterStatusInstalling,
+	models.ClusterStatusFinalizing,
+	models.ClusterStatusInstalled,
+}
+
+// The order of this list matters. It needs to follow a host happy-flow
+var hostValidSortedStatuses = [...]string{
+	models.HostStatusDiscovering,
+	models.HostStatusKnown,
+	models.HostStatusInsufficient,
+	models.HostStatusPreparingForInstallation,
+	models.HostStatusPendingForInput,
+	models.HostStatusInstalling,
+	models.HostStatusInstallingInProgress,
+	models.HostStatusInstallingPendingUserAction,
+	models.HostStatusResettingPendingUserAction,
+	models.HostStatusInstalled,
+	models.HostStatusResetting,
```

Comment:
We can move from resetting to known no?

Reply:
States are not serial. we can move between multiple states  back and forth.

Reply:
So this isn't a good solution.